hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/StorageContainerServiceProviderImpl.java

```diff
@@ -160,15 +167,39 @@ public DBCheckpoint getSCMDBSnapshot() {
             ".tar.gz");
 
     try {
-      SecurityUtil.doAsLoginUser(() -> {
-        try (InputStream inputStream = reconUtils.makeHttpCall(
-            connectionFactory, getScmDBSnapshotUrl(),
-            isOmSpnegoEnabled()).getInputStream()) {
-          FileUtils.copyInputStreamToFile(inputStream, targetFile);
+      if (!SCMHAUtils.isSCMHAEnabled(configuration)) {
+        SecurityUtil.doAsLoginUser(() -> {
+          try (InputStream inputStream = reconUtils.makeHttpCall(
+              connectionFactory, getScmDBSnapshotUrl(),
+              isOmSpnegoEnabled()).getInputStream()) {
+            FileUtils.copyInputStreamToFile(inputStream, targetFile);
+          }
+          return null;
+        });
+      } else {
+        List<String> ratisRoles = scmClient.getScmInfo().getRatisPeerRoles();
+        for (String ratisRole: ratisRoles) {
+          String[] role = ratisRole.split(":");
+          if(role[2].equals(RaftProtos.RaftPeerRole.LEADER.toString())) {
+            String hostAddress = role[4].trim();
+            int grpcPort = configuration.getInt(
+                ScmConfigKeys.OZONE_SCM_GRPC_PORT_KEY,
+                ScmConfigKeys.OZONE_SCM_GRPC_PORT_DEFAULT);
+
+            try (SCMSnapshotDownloader downloadClient =
+                 new InterSCMGrpcClient(hostAddress, grpcPort,
+                 configuration, new SCMCertificateClient(
+                 new SecurityConfig(configuration)))) {
+              downloadClient.download(targetFile.toPath()).get();
+            } catch (ExecutionException | InterruptedException e) {
+              LOG.error("Rocks DB checkpoint downloading failed", e);
+              throw new IOException(e);
+            }
+            LOG.info("Downloaded SCM Snapshot from Leader SCM");
```

Comment:
Let's add a simalar message to non-HA code path as well.

Reply:
Updated the PR with the suggestion